### PR TITLE
UDF: Add deregister method.

### PR DESCRIPTION
**Pull request by JohnMoutafis**
- Adds deregister method.
-  Also refactors delete method to work with URI.

### CODE REVIEW

File: src/tiledb/cloud/udf.py [hunk 1]
```diff
@@ -550,14 +550,26 @@ def unshare(name=None, namespace=None, async_req=False):
 """
 
 
-def delete(name, namespace, async_req=False):
+def delete(
```
Comment (Shelnutt2) [inline src/tiledb/cloud/udf.py:553]: I believe this order change breaks the compatibility of the API? Users that are passing by position today would have breakage?

Comment (JohnMoutafis) [reply src/tiledb/cloud/udf.py:553]: I made the change to start keeping a more uniform style on those methods going forward.

What I can do is rename this method to `delete_new` and reinstate the previous `delete` method to work as expected with a deprecation warning.

File: src/tiledb/cloud/udf.py [hunk 2]
```diff
@@ -550,14 +551,40 @@ def unshare(name=None, namespace=None, async_req=False):
...
+    DEPRECATION WARNING: This method will be replaced by delete_v2
+                         as delete from version 0.12.17 onwards
+    """
+    warning(
+        DeprecationWarning(
+            "This method will be replaced by the `delete_v2` method,",
+            "which will be renamed to `delete` from version 0.12.17.",
+        ),
+    )
+    delete_v2(name=name, namespace=namespace, async_req=async_req)
```
Comment (JohnMoutafis) [inline src/tiledb/cloud/udf.py:565]: @Shelnutt2 do you agree with that approach?

File: src/tiledb/cloud/udf.py (hunk 1 above)
Comment (Shelnutt2) [reply src/tiledb/cloud/udf.py]: How about we add the uri parameter, to this function, deprecate the existing parameters and warn users. Then we can a avoid the `v2` function name, support compatibility for now and give us a clean future API?

Comment (JohnMoutafis) [reply src/tiledb/cloud/udf.py]: @ihnorton both deprecation approaches have their pros and cons.
We will highly value your input on which method to follow going forward (many methods will eventually be refactored)

Comment (ihnorton) [reply src/tiledb/cloud/udf.py]: We could make the change but fall back to the existing (interpret as `name`) behavior if the argument doesn't start with `tiledb://` -- and print a deprecation warning.

Comment (JohnMoutafis) [reply src/tiledb/cloud/udf.py:554]: According to @ihnorton suggestion, I modified the method as shown here.
The `namespace` argument will be available for backwards compatibility until version `0.12.17` and the method will throw a deprecation warning if the `uri` is not valid.
FYI @Shelnutt2

File: src/tiledb/cloud/udf.py [hunk 3]
```diff
@@ -550,14 +551,32 @@ def unshare(name=None, namespace=None, async_req=False):
...
-    :param async_req: return future instead of results for async support
+
+    :param uri: TileDB URI of the udf, defaults to None.
+    :param namespace: namespace the udf belongs to, defaults to None.
+        DEPRECATION WARNING: Will be deprecate from version 0.12.17
+    :param async_req: Return future instead of results for async support
     :return: deleted udf details
     """
+    try:
+        namespace, name = utils.split_uri(uri)
```
Comment (ihnorton) [inline src/tiledb/cloud/udf.py:567]: Does this work correctly for `tiledb://<namespace>/s3://...` URIs?

**Review by ihnorton — APPROVED** [review]
One question but otherwise LGTM.

File: src/tiledb/cloud/udf.py (hunk 3 above)
Comment (JohnMoutafis) [reply src/tiledb/cloud/udf.py:567]: Sort of.
- `namespace` will be correct
- `name` will be the whole `s3://...` path

**Pull request merged**